test/replicate.test.js

```diff
@@ -4,293 +4,197 @@ const assert = require('assert')
 const OrbitDB = require('../src/OrbitDB')
-const config = require('./utils/config')
-const startIpfs = require('./utils/start-ipfs')
-const waitForPeers = require('./utils/wait-for-peers')
+
+// Include test utilities
+const {
+  config,
+  startIpfs,
+  stopIpfs,
+  testAPIs,
+  connectPeers,
+  waitForPeers,
+  MemStore,
+} = require('./utils')
 
 const dbPath1 = './orbitdb/tests/replication/1'
 const dbPath2 = './orbitdb/tests/replication/2'
 const ipfsPath1 = './orbitdb/tests/replication/ipfs/1'
 const ipfsPath2 = './orbitdb/tests/replication/ipfs/2'
 
-const MemStore = require('./utils/mem-store')
-
-describe('orbit-db - Replication', function() {
-  this.timeout(config.timeout * 2)
-
-  let ipfs1, ipfs2, orbitdb1, orbitdb2, db1, db2
-
-  describe('two peers', function() {
-    let timer
-    let options
-
-    beforeEach(async () => {
-      clearInterval(timer)
-
-      config.daemon1.repo = ipfsPath1
-      config.daemon2.repo = ipfsPath2
-      rmrf.sync(config.daemon1.repo)
-      rmrf.sync(config.daemon2.repo)
-      rmrf.sync(dbPath1)
-      rmrf.sync(dbPath2)
-      ipfs1 = await startIpfs(config.daemon1)
-      ipfs2 = await startIpfs(config.daemon2)
-      // Use memory store for quicker tests
-      const memstore = new MemStore()
-      ipfs1.object.put = memstore.put.bind(memstore)
-      ipfs1.object.get = memstore.get.bind(memstore)
-      ipfs2.object.put = memstore.put.bind(memstore)
-      ipfs2.object.get = memstore.get.bind(memstore)
-      // Connect the peers manually to speed up test times
-      await ipfs2.swarm.connect(ipfs1._peerInfo.multiaddrs._multiaddrs[0].toString())
-      await ipfs1.swarm.connect(ipfs2._peerInfo.multiaddrs._multiaddrs[0].toString())
-      orbitdb1 = new OrbitDB(ipfs1, dbPath1)
-      orbitdb2 = new OrbitDB(ipfs2, dbPath2)
-
-      options = { 
-        // Set write access for both clients
-        write: [
-          orbitdb1.key.getPublic('hex'), 
-          orbitdb2.key.getPublic('hex')
-        ],
-      }
-
-      options = Object.assign({}, options, { path: dbPath1 })
-      db1 = await orbitdb1.eventlog('replication-tests', options)
-      // Set 'sync' flag on. It'll prevent creating a new local database and rather
-      // fetch the database from the network
-//      options = Object.assign({}, options, { path: dbPath2, sync: true })
-//      db2 = await orbitdb2.eventlog(db1.address.toString(), options)
-
-//      assert.equal(db1.address.toString(), db2.address.toString())
-
-//      await waitForPeers(ipfs1, [orbitdb2.id], db1.address.toString())
-//      await waitForPeers(ipfs2, [orbitdb1.id], db1.address.toString())
-    })
-
-    afterEach(async () => {
-      clearInterval(timer)
-      options = {}
-
-      if (db1)
-        await db1.drop()
-      
-      if (db2) 
-        await db2.drop()
-
-      if(orbitdb1) 
-        await orbitdb1.stop()
+Object.keys(testAPIs).forEach(API => {
+  describe.only(`orbit-db - Replication (${API})`, function() {
```

Comment:
Not sure if this is still WIP but there's a `.only` here that will avoid other tests from running on CI